### PR TITLE
fix(comments): Created users_permission_user field to Comment type

### PR DESCRIPTION
**Pull request by RETprojects**

### CODE REVIEW

**Comment by becevka** [comment]
@RETprojects I agree, there’s an issue with having comment author as a plain text. However, I suggest working with @zar92 on solution that will be helpful for the long term approach. Also, we might need to consider migration for existing comments. @chungthuang do we have comments in prod database?

**Comment by zar92** [comment]
Which ticket is this related to? @becevka @RETprojects
I am a little unsure if the Backend team is the owner of this issue. We are creating notifications for existing comments. If there is a document where I can see the existing issue, please link it here.

**Comment by RETprojects** [comment]
@zar92 This is mostly related to [#1684](https://github.com/dev-launchers/dev-launchers-platform/issues/1684), [US - DEV] Link Users to Comments with Database Relation. This would help to track user activity on the workshopping page, like adding the idea owner tag in the comment feed.

**Comment by becevka** [comment]
@RETprojects Adding one to one relationship definitely will not work as one user might want to create multiple comments. Also, your branch is quite outdated and has conflicts. 
I propose to close this PR and we will handle it in scope of https://github.com/dev-launchers/strapiv4/issues/158

cc @zar92

**Comment by chungthuang** [comment]
We already have some comments in production.

**Comment by dbradham** [comment]
> @RETprojects Adding one to one relationship definitely will not work as one user might want to create multiple comments. Also, your branch is quite outdated and has conflicts. I propose to close this PR and we will handle it in scope of #158
> 
> cc @zar92

Thanks @becevka that sounds like a good plan. Do we have a timeline to close #158 ? This blocks ideaspace from making any progress on comments which are a top priority for us.

**Comment by dbradham** [comment]
> @RETprojects Adding one to one relationship definitely will not work as one user might want to create multiple comments. Also, your branch is quite outdated and has conflicts. I propose to close this PR and we will handle it in scope of #158
> 
> cc @zar92

Hi,

Both of these issues have been addressed. Can you please review again or let us know a timeline for the alternative you have proposed? Like I mentioned yesterday, this is a top priority for IdeaSpace development team. 

Thanks!

**Comment by zar92** [comment]
@dbradham, This ticket will be prioritized in the next sprint. Based on complexity, we may need to confirm completion date later on. But usually it should be solved within the next sprint.

**Comment by dbradham** [comment]
> @dbradham, This ticket will be prioritized in the next sprint. Based on complexity, we may need to confirm completion date later on. But it should be solved within the next sprint.



> @dbradham, This ticket will be prioritized in the next sprint. Based on complexity, we may need to confirm completion date later on. But it should be solved within the next sprint.

It looks like no one is assigned to work on this task. Can you please let me know who is working on this change? 

Also, the #158 ticket only mentions the authors of idea's, not comments. Can you explain to use how this change is included the in scope of #158 ?

**Comment by dbradham** [comment]
> > @dbradham, This ticket will be prioritized in the next sprint. Based on complexity, we may need to confirm completion date later on. But it should be solved within the next sprint.
> 
> It looks like no one is assigned to work on this task. Can you please let me know who is working on this change?
> 
> Also, the #158 ticket only mentions the author's are misattributed, but nothing about the underlying data schema which is what we are addressing in this PR. Can you explain to use how this change is included the in scope of #158 ? @zar92

@becevka friendly nudge here. We've now been blocked on development for 2 weeks. Do you have a timeline to resolve, or can this PR be approved?

**Comment by zar92** [comment]
@dbradham, I saw that @becevka had a discussion about this issue with Remi and Yomi. And he tagged you in that discussion. Going forward, let's put everything in the discord channel.

**Comment by becevka** [comment]
@RETprojects Also, please prefix your commits and PR name with `fix(comments)`

**Comment by dbradham** [comment]
Thanks @becevka! Remi has made the changes discussed last week. Please let us know if you have any other requested changes here.

File: types/generated/contentTypes.d.ts [hunk 1]
```diff
@@ -640,8 +640,13 @@ export interface PluginUsersPermissionsUser extends Schema.CollectionType {
       'oneToOne',
       'api::profile.profile'
     >;
-    discordId: Attribute.String & Attribute.Private;
-    googleId: Attribute.String & Attribute.Private;
+    time_capsules: Attribute.Relation<
```
Comment (becevka) [inline types/generated/contentTypes.d.ts:643]: We have removed Time Capsule in https://github.com/dev-launchers/strapiv4/pull/146/files#diff-7d06fadd606542cc6ab66af65e7560f74b247e444f5abb8f347063f53a2ac30aL643

File: types/generated/contentTypes.d.ts [hunk 2]
```diff
@@ -1482,6 +1493,49 @@ export interface ApiTeamMembershipTeamMembership extends Schema.CollectionType {
   };
 }
 
+export interface ApiTimeCapsuleTimeCapsule extends Schema.CollectionType {
```
Comment (becevka) [inline types/generated/contentTypes.d.ts:1496]: Additional extra off. Maybe makes sense to merge in latest dev?

**Comment by dbradham** [comment]
@becevka @oriyomibadmus can you please let Remi know if there are any outstanding items here to address? Thank you!

File: src/api/comment/content-types/comment/lifecycles.js [hunk 3]
```diff
@@ -1,5 +1,12 @@
 module.exports = {
 
+  beforeCreate(event) {
+    // Set author to the user sending the request
+    const ctx = strapi.requestContext.get();
+    event.params.data.author = ctx.state.user;
```
Comment (dbradham) [inline src/api/comment/content-types/comment/lifecycles.js:6]: what is the benefit of using this lifecycle, as opposed to setting the current user data on the front end which is the current implementation?

Comment (becevka) [reply src/api/comment/content-types/comment/lifecycles.js:6]: It should be event.params.data.authorId = ctx.state.user. 
@dbradham the idea is to not allow someone pass arbitrary authorId from frontEnd, instead authorId is populated by backend using current user.

@oriyomibadmus Based on what you've discovered, if migrations are running before schema sync up, maybe there's no benefit on adding new column we can just migrate data into existing and change its type in schema sync. Could you please experiment with migrations locally?

File: types/generated/contentTypes.d.ts (hunk 2 above)
Comment (becevka) [reply types/generated/contentTypes.d.ts:1496]: @RETprojects this change is still extra.

File: src/api/comment/content-types/comment/lifecycles.js (hunk 3 above)
Comment (dbradham) [reply src/api/comment/content-types/comment/lifecycles.js]: Thanks @becevka, but isn't this outside the scope of Remi's change? It looks more related to #158 and not his ticket. (https://app.zenhub.com/workspaces/-ideaspace-team-6350185d9843157427cc900b/issues/gh/dev-launchers/dev-launchers-platform/1684)

I'm confused on why this minor change is NOT approved, so Remi can continue to develop his other ticket? (https://app.zenhub.com/workspaces/-ideaspace-team-6350185d9843157427cc900b/issues/gh/dev-launchers/dev-launchers-platform/748)

Comment (dbradham) [reply src/api/comment/content-types/comment/lifecycles.js]: @becevka if you cannot approve this change, can you please schedule a call with me to discuss further? I am available any weeknight after 7pm Central US Time. Or on the weekends I can make myself available outside of 10-11am Central US Time.

File: types/generated/contentTypes.d.ts [hunk 4]
```diff
@@ -1482,6 +1493,49 @@ export interface ApiTeamMembershipTeamMembership extends Schema.CollectionType {
   };
 }
 
+// export interface ApiTimeCapsuleTimeCapsule extends Schema.CollectionType {
```
Comment (becevka) [inline types/generated/contentTypes.d.ts:1496]: @RETprojects could you please remove the extra lines.

File: types/generated/contentTypes.d.ts [hunk 5]
```diff
@@ -1516,6 +1570,7 @@ declare module '@strapi/types' {
       'api::save-idea.save-idea': ApiSaveIdeaSaveIdea;
       'api::sendmail.sendmail': ApiSendmailSendmail;
       'api::team-membership.team-membership': ApiTeamMembershipTeamMembership;
+      'api::time-capsule.time-capsule': ApiTimeCapsuleTimeCapsule;
```
Comment (becevka) [inline types/generated/contentTypes.d.ts:1573]: @RETprojects Could you please remove extra line. We have already removed TimeCapsule

File: src/api/comment/content-types/comment/lifecycles.js (hunk 3 above)
Comment (becevka) [reply src/api/comment/content-types/comment/lifecycles.js]: @dbradham Remi was interested in getting this code change through.
 The only note left is extra code which needs to be removed. 
 Also, this change kinda renders changes on Front End unnecessary as FrontEnd will not need to provide author.

**Review by becevka — APPROVED** [review]

**Pull request merged**